astropy/cosmology/core.py

```diff
@@ -66,11 +202,37 @@ class Cosmology(metaclass=ABCMeta):
     read = UnifiedReadWriteMethod(CosmologyRead)
     write = UnifiedReadWriteMethod(CosmologyWrite)
 
+    # Parameters
+    __parameters__ = ()
+    _order_params_by_signature = False
+
     def __init_subclass__(cls):
         super().__init_subclass__()
 
+        # registry of Cosmology's subclasses
         _COSMOLOGY_CLASSES[cls.__qualname__] = cls
 
+        # -------------------
+        # Parameters
+
+        # names of Parameters  (filtered from parent class. maintains order)
+        parameters = [
+            n for n in cls.__parameters__  # ← from Parameters self-registering
```

Comment:
It does feel like one could just look for extra parameters in `cls.__dict__` here. It would have the advantage of an attribute not having effect outside of itself.

Reply:
See https://github.com/astropy/astropy/pull/12136#discussion_r701538764.